test/es-module/test-esm-loader-entry-url.mjs

```diff
@@ -0,0 +1,85 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+describe('--entry-url', { concurrency: true }, () => {
+  it('should reject loading absolute path that contains %', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--entry-url',
+        fixtures.path('es-modules/test-esm-double-encoding-native%20.mjs'),
+      ]
+    );
+
+    assert.match(stderr, /ERR_MODULE_NOT_FOUND/);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should support loading absolute Unix path properly encoded', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--entry-url',
+        fixtures.fileURL('es-modules/test-esm-double-encoding-native%20.mjs').pathname,
+      ]
+    );
+
+    assert.match(stderr, /--entry-url is an experimental feature/);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should support loading absolute URLs', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--entry-url',
+        fixtures.fileURL('printA.js'),
+      ]
+    );
+
+    assert.match(stderr, /--entry-url is an experimental feature/);
+    assert.match(stdout, /^A\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should support loading relative URLs', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--entry-url',
+        './printA.js?key=value',
```

Comment:
```suggestion
        './printA.js?key=value#hashstring',
```
Also it would be great to test if `import.meta.url` includes the `search` and `hash` parts, since it's easy to accidentally strip these in the resolver for local files.
Doesn't necessarily have to be in this PR, can be follow-up.